### PR TITLE
fix: add back support for imap4flags

### PR DESCRIPTION
**Pull request by russell**
Add back the support for imap4flags, and also add a test for them.

During the course of investigating this error some of the error messages coming out of the extensions class didn't provide enough information, so i have added any context information i could into those too.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -66,6 +66,9 @@
             'require = sifter.commands.require:CommandRequire',
             'set = sifter.commands.variables:CommandSet',
             'stop = sifter.commands.stop:CommandStop',
+            'imap4flags.setflag = sifter.commands.imap4flags:CommandSetFlag',
```
Comment (russell) [inline setup.py:69]: @manfred-kaiser i'm not 100% on the best way to declare these. I put a dot in the name because there are 3 commands to the imap4flags

Comment (russell) [reply setup.py:69]: i think these names are just an annotation and aren't used in the logic

File: tests/test_evaluation.py [hunk 2]
```diff
@@ -29,6 +29,11 @@ def test_evaulation():
         ("evaluation_1.msg", "evaluation_6.rules", [('reject', 'I do not accept messages from this address.')]),
         ("evaluation_1.msg", "evaluation_7.rules", [('reject', 'I do not accept messages from/* this is\nnot a comment */this address.\n.\n')]),
         ("evaluation_1.msg", "evaluation_8.rules", [('keep', None)]),
+        ("evaluation_1.msg", "evaluation_9.rules", [('setflag', ['test-set']),
+                                                    ('addflag', ['test-add']),
+                                                    ('removeflag', ['test-remove']),
+                                                    ('keep', None),
```
Comment (russell) [inline tests/test_evaluation.py:35]: there is a duplicate keep here, i'm not sure where that is coming from.  it's not from the imap code

**Review by manfred-kaiser — APPROVED** [review]
Seems ok👍

**Pull request merged**